model/lifecycle.go

```diff
@@ -1633,7 +1635,8 @@ func addNewBuilds(ctx context.Context, creationInfo TaskCreationInfo, existingBu
 			},
 		)
 	}
-	if err = task.UpdateSchedulingLimit(creationInfo.Version.Author, creationInfo.Version.Requester, len(newActivatedTaskIds), true); err != nil {
+	numTasksModified := numEstimatedActivatedGeneratedTasks + len(newActivatedTaskIds)
```

Comment:
Also, does this count dependencies that are newly activated by the newly-generated tasks, or is it just the tasks that will be directly activated? I see `ActivateTasks` counts dependencies updated as tasks scheduled, but I don't see it here/in patch finalization.

Reply:
For task generation, that should be taken care of when [addNewTasksToExistingBuilds](https://github.com/evergreen-ci/evergreen/blob/main/model/lifecycle.go#L1758-L1764) runs. I don't believe it's necessary to check for indirectly activated tasks during patch finalization since finalization only creates new tasks.